### PR TITLE
Fix linkedin flow with no cookies

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Fix LinkedIn auth button flow when using a browser with 3rd party cookies blocked (Safari / Chrome incognito).

Adds a 5th authenticate-by-LTI-state method to our supported methods, to allow auth by iframe inside CustomContent (in this case, the Hustle to Career Project).

### Motivation and Context

Task https://app.asana.com/0/1174274412967132/1199407082341983/f

### Screenshots (if appropriate):

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: spec/feature/linked_in_authorization/linked_in_authorize_spec.rb [hunk 1]
```diff
@@ -5,31 +5,43 @@
 
 RSpec.feature 'Authorize LinkedIn access to user data', :type => :feature do
   let(:canvas_user_id) { 1717 }
-  let!(:user) { create :linked_in_user, canvas_user_id: canvas_user_id }
+  let(:section) { create(:section) }
+  let!(:user) { create :linked_in_user, canvas_user_id: canvas_user_id, section: section }
   let!(:lti_launch) { create( :lti_launch_assignment, canvas_user_id: canvas_user_id) }
+  let(:linkedin_project) { create(:linkedin_project) }
+  let(:linkedin_project_version) { create(:project_version, body: linkedin_project.body, custom_content: linkedin_project) }
+  let(:course_project_version) { create(:course_project_version, course: section.course, custom_content_version: linkedin_project_version) }
```
Comment (rshipp) [inline spec/feature/linked_in_authorization/linked_in_authorize_spec.rb:13]: there's no way to set the referer header in selenium, so instead we just create a project with a real linkedin button iframe, and that sets a real referer header.

Comment (jessouyangbraven) [reply spec/feature/linked_in_authorization/linked_in_authorize_spec.rb:13]: nit: add in comment in code? I can see myself fighting Selenium trying to get this work and not realizing it's happening automatically by creating the project.

**Review by jessouyangbraven — APPROVED** [review]
LGTM! Thanks for fixing this!

**Review by sadleb — APPROVED** [review]
lgtm

**Pull request merged**